integrated_channels/moodle/transmitters/content_metadata.py

```diff
@@ -21,6 +34,105 @@ def __init__(self, enterprise_configuration, client=MoodleAPIClient):
+                        self.enterprise_configuration.enterprise_customer.name,
+                        self.enterprise_configuration.channel_code(),
+                        exc.message,
+                        exc.status_code
+                    )
+                    LOGGER.exception(exc)
+            else:
+                self._create_transmissions(chunk)
+
+    def _transmit_update(self, channel_metadata_item_map, transmission_map, chunk_size=None, count=0):  # pylint: disable=arguments-differ
+        """
+        Transmit content metadata update to integrated channel.
+        """
+        chunk_size = chunk_size if chunk_size else self.enterprise_configuration.transmission_chunk_size
+        chunk_items = chunks(channel_metadata_item_map, chunk_size)
+        transmission_limit = settings.INTEGRATED_CHANNELS_API_CHUNK_TRANSMISSION_LIMIT.get(
+            self.enterprise_configuration.channel_code()
+        )
+        for chunk in islice(chunk_items, transmission_limit):
+            serialized_chunk = self._serialize_items(list(chunk.values()))
+            try:
+                self.client.update_content_metadata(serialized_chunk)
+            except ClientError as exc:
+                if exc.status_code == 414 and count <= 1:
+                    count += 1
+                    sleep(5)
+                    self._transmit_update(channel_metadata_item_map, transmission_map, chunk_size=LOW_CHUNK_SIZE, count=count)
+                else:
+                    LOGGER.error(
+                        'Failed to update [%s] content metadata items for integrated channel [%s] [%s]. '
+                        'Task failed with message [%s] and status code [%s]',
+                        len(chunk),
+                        self.enterprise_configuration.enterprise_customer.name,
+                        self.enterprise_configuration.channel_code(),
+                        exc.message,
+                        exc.status_code
+                    )
+                    LOGGER.exception(exc)
+            else:
+                self._update_transmissions(chunk, transmission_map)
+
+    def _transmit_delete(self, channel_metadata_item_map, chunk_size=None, count=0):  # pylint: disable=arguments-differ
+        """
+        Transmit content metadata deletion to integrated channel.
+        """
+        chunk_size = chunk_size if chunk_size else self.enterprise_configuration.transmission_chunk_size
+        chunk_items = chunks(channel_metadata_item_map, chunk_size)
+        transmission_limit = settings.INTEGRATED_CHANNELS_API_CHUNK_TRANSMISSION_LIMIT.get(
+            self.enterprise_configuration.channel_code()
+        )
+        for chunk in islice(chunk_items, transmission_limit):
+            serialized_chunk = self._serialize_items(list(chunk.values()))
+            try:
+                self.client.delete_content_metadata(serialized_chunk)
+            except ClientError as exc:
+                if exc.status_code == 414 and count <= 1:
```

Comment:
I am a bit unclear on this change and the description change combined. Even despite the description change you made, will we expect to hit 414s? I am wondering if we also considered the alternative which is to send truncated descriptions and avoid the problem (basically ensure max desc sizes are set by us). I think clicking on edX link can take us to the full description website anyway, is having the full description in moodle a must? may want to discuss this one in team channel... it would make things simpler for us (and thanks Moodle for GET apis!)

Reply:
Also without limiting the description, even if we have a chunk size of 1, _if_ the description *does* contain non-latin characters, we can potentially hit the 414 on a single item.

Reply:
aargh yeah that does limit us quite a bit then hah! damn api that uses query strings...

Since it looks like worst case we can still hit 414 even with 1 chunk size, I will make one last pitch to see if we can get away with truncating the long description until we do fall within limit. Basically we still send it but truncated. I think we can convince product folks that is okay to do. Basically [long text with ... at the end]. Since edX will show full description anyway.  

But having said that, your idea of using the short only description also is a variation of that anyway so I am also okay with this idea.

Reply:
I mean that's why I'm using shorter values and that "use short_description if full description is too long" logic exists in other integrated_channels implementations, so it's not a unique hack here. I am just being safe around the short_description as well since it **technically** is a "longtext" field which can be up to 4gb. We don't have any data in our production database that goes anywhere remotely near this length, but it _is_ possible.

Reply:
Based on this prior art, it seemed like the consensus was that a short description was better than arbitrarily cutting off the full description.